test/assert.js

```diff
@@ -74,6 +74,26 @@ assert.holdsItem = function (pokemon, message) {
 	});
 };
 
+assert.trapped = function (fn) {
+	assert.throws(() => fn(), Error, "[Invalid choice] Can't switch: The active Pokémon is trapped");
+};
+
+assert.cantMove = function (fn, pokemon, move) {
+	if (pokemon && move) {
+		assert.throws(() => fn(), Error, new RegExp(`[Invalid choice] Can't move:.*${pokemon}.*${move}`, 'i'));
+	} else {
+		assert.throws(() => fn(), Error, new RegExp(`[Invalid choice] Can't move:`));
+	}
+};
+
+assert.cantUndo = function (fn) {
+	assert.throws(() => fn(), Error, new RegExp("[Invalid choice] Can't undo:"));
+};
```

Comment:
Gotta escape the brackets

Reply:
```
Pokemon-Showdown/test/assert.js
  83:48  error  Unnecessary escape character: \[  no-useless-escape
  83:64  error  Unnecessary escape character: \]  no-useless-escape
  85:48  error  Unnecessary escape character: \[  no-useless-escape
  85:64  error  Unnecessary escape character: \]  no-useless-escape
  90:47  error  Unnecessary escape character: \[  no-useless-escape
  90:63  error  Unnecessary escape character: \]  no-useless-escape
```

Reply:
No, `eslint` is correct. Your escapes are useless. `"\["` means the same thing as `"["` – it's a string passed to a regex construct, not a real regex. You need to do `"\\[`.